tests/rules/javascript/test.deprecated_entities.js

```diff
@@ -0,0 +1,77 @@
+import { VALIDATION_WARNING, DEPRECATED_ENTITIES } from 'const';
+import { singleLineString } from 'utils';
+import JavaScriptScanner from 'scanners/javascript';
+
+describe('no_document_write', () => {
+  for (let entity of DEPRECATED_ENTITIES) {
+    it(`should warn about using ${entity.object}.${entity.property}()`, () => {
+      var code = `${entity.object}.${entity.property}();`;
+      var jsScanner = new JavaScriptScanner(code, 'badcode.js');
+
+      return jsScanner.scan()
+        .then((validationMessages) => {
+          assert.equal(validationMessages.length, 1);
+          assert.equal(validationMessages[0].code,
+                       entity.error.code);
+          assert.equal(validationMessages[0].type, VALIDATION_WARNING);
+        });
+    });
+
+    it(`should know when a variable references ${entity.object}`, () => {
+      var code = singleLineString`var foo = ${entity.object};
+        foo.${entity.property}();`;
+      var jsScanner = new JavaScriptScanner(code, 'badcode.js');
+
+      return jsScanner.scan()
+        .then((validationMessages) => {
+          assert.equal(validationMessages.length, 1);
+          assert.equal(validationMessages[0].code,
+                       entity.error.code);
+          assert.equal(validationMessages[0].type, VALIDATION_WARNING);
+        });
+    });
+
+    it('should not warn about using other member functions', () => {
+      var code = `${entity.object}.doNothing();`;
+      var jsScanner = new JavaScriptScanner(code, 'badcode.js');
+
+      return jsScanner.scan()
+        .then((validationMessages) => {
+          assert.equal(validationMessages.length, 0);
+        });
+    });
+
+    it(`should not warn about calling ${entity.property} in general`, () => {
+      var code = `foo.${entity.property}();`;
+      var jsScanner = new JavaScriptScanner(code, 'badcode.js');
+
+      return jsScanner.scan()
+        .then((validationMessages) => {
+          assert.equal(validationMessages.length, 0);
+        });
+    });
+
+    it(singleLineString`should not warn about calling ${entity.property} to a
```

Comment:
Seems to be getting confused by the template string's 2nd line indent. Maybe you could try using vars in the loop (before the test-cases) to shorten the template subs vars? It would be nice to avoid needing `singleLineString` for a test description.